framework/src/modules/http_api/controllers/node.js

```diff
@@ -15,13 +15,65 @@
 'use strict';
 
 const _ = require('lodash');
+const { getAddressFromPublicKey } = require('@liskhq/lisk-cryptography');
 const checkIpInList = require('../helpers/check_ip_in_list');
 const apiCodes = require('../api_codes');
 const swaggerHelper = require('../helpers/swagger');
 
 let library;
 let epochTime;
-let fees;
+
+function _filterTransactions(transactions, filters) {
+	const { limit, offset, sort } = filters;
+	let filteredTransactions = transactions;
+	if (filters.id) {
+		filteredTransactions = filteredTransactions.filter(
+			transaction => transaction.id === filters.id,
+		);
+	}
+	if (filters.senderId || filters.senderPublicKey) {
+		filteredTransactions = filteredTransactions.filter(transaction => {
+			if (filters.senderId) {
+				return (
+					getAddressFromPublicKey(transaction.senderPublicKey) ===
+					filters.senderId
+				);
+			}
+			return transaction.senderPublicKey === filters.senderPublicKey;
+		});
+	}
+	if (filters.type !== null && filters.type !== undefined) {
+		filteredTransactions = filteredTransactions.filter(
+			transaction => transaction.type === filters.type,
+		);
+	}
```

Comment:
do we need the null and undefined check? as the swagger takes care of validation, if it's not taken care it should be done there.

Reply:
it is filter, where it might or might not exist, and since this is number, it should check `null` and `undefined`

Reply:
then why not `Number.isSafeInteger(filters.type)`?